### PR TITLE
Feature/return df

### PR DESCRIPTION
**Pull request by kevindougherty-noaa**
Addresses #75

### CODE REVIEW

**Comment by kevindougherty-noaa** [comment]
@CoryMartin-NOAA still need to go through the other files for `pycodestyle` to pass, however `diags.py` passes which is the only file I changed.

**Comment by CoryMartin-NOAA** [comment]
@kevindougherty-noaa can you explain to @ADCollard and me why this PR is needed? I'm sure it is a method vs attribute thing but it's not clear to me at the moment.

**Comment by kevindougherty-noaa** [comment]
@CoryMartin-NOAA @ADCollard Sure thing. In our regional DA meeting last week, Ting and Shun mentioned interest in being able to have the entire dataframe returned. I could be mistaken and maybe they meant returning the dataframe after being indexed.

File: pyGSI/diags.py [hunk 1]
```diff
@@ -643,6 +643,10 @@ def get_lat_lon(self, obsid=None, subtype=None, station_id=None,
 
                 return lats, lons
 
+    def return_df(self):
+        """Returns working dataframe."""
+        return self.data_df
+
```
Comment (aerorahul) [inline pyGSI/diags.py]: Why can't you add this method in `Class GSIdiag` since that is the class from which all other inherit from?

**Review by aerorahul — CHANGES_REQUESTED** [review]
See comment.

**Comment by CoryMartin-NOAA** [comment]
@kevindougherty-noaa yes I remember that, so I guess you're saying this is so one can have numpy arrays if they want, and a Pandas dataframe through the `.return_df()` method? I thought the issue you linked would basically imply that the read method would have an argument like `read_data(return_df=True)` and then:
```
if return_df:
    return self.df
else:
   return data
```
or whatever the above pseudo code should be

**Comment by kevindougherty-noaa** [comment]
@CoryMartin-NOAA you are right, that is what should be done. I will fix.

File: pyGSI/diags.py [hunk 2]
```diff
@@ -224,52 +227,64 @@ def get_data(self, diag_type, obsid=None, subtype=None, station_id=None,
...
-                    monitored_data = self._query_diag_type(
-                        monitored_df, diag_type, bias_correction)
-
-                    data = {'assimilated': assimilated_data,
-                            'rejected': rejected_data,
-                            'monitored': monitored_data
-                            }
-
-                    return data
+				if return_df:
```
Comment (CoryMartin-NOAA) [inline pyGSI/diags.py]: @kevindougherty-noaa is this indentation correct?

File: pyGSI/diags.py [hunk 3]
```diff
@@ -224,52 +227,64 @@ def get_data(self, diag_type, obsid=None, subtype=None, station_id=None,
...
-
-                    u = {'assimilated': u_assimilated,
-                         'rejected': u_rejected,
-                         'monitored': u_monitored}
-                    v = {'assimilated': v_assimilated,
-                         'rejected': v_rejected,
-                         'monitored': v_monitored}
-
-                    return u, v
+                if return_df:
```
Comment (CoryMartin-NOAA) [inline pyGSI/diags.py:230]: so for non data frame, it returns a dict of dicts, for u,v. For the data frame case, are both u and v inside the data frame?

Comment (kevindougherty-noaa) [reply pyGSI/diags.py:230]: Yes, both u and v are in the dataframe. All the variables will be included.

File: pyGSI/diags.py [hunk 4]
```diff
@@ -224,52 +227,64 @@ def get_data(self, diag_type, obsid=None, subtype=None, station_id=None,
...
-                        indexed_df, diag_type, bias_correction)
+                    if self.variable == 'uv':
+                        u, v = self._query_diag_type(
+                            indexed_df, diag_type, bias_correction)
+
+                        return u, v
+
+                    else:
+                        data = self._query_diag_type(
+                            indexed_df, diag_type, bias_correction)
```
Comment (aerorahul) [inline pyGSI/diags.py]:
```suggestion
                    return self._query_diag_type(
                            indexed_df, diag_type, bias_correction)
```

File: pyGSI/diags.py [hunk 5]
```diff
@@ -224,52 +227,64 @@ def get_data(self, diag_type, obsid=None, subtype=None, station_id=None,
...
 
-                    return data
+                        return data
```
Comment (aerorahul) [inline pyGSI/diags.py]:
```suggestion
```

**Review by aerorahul — CHANGES_REQUESTED** [review]
Just a quick glance. 
See if you notice these as well.

File: pyGSI/diags.py (hunk 4 above)
Comment (CoryMartin-NOAA) [reply pyGSI/diags.py]: This is a good suggestion

Comment (kevindougherty-noaa) [reply pyGSI/diags.py]: I like it, thanks @aerorahul.

**Comment by kevindougherty-noaa** [comment]
Based on conversations with @CoryMartin-NOAA and @aerorahul, I am going to close this PR and develop `diags.py` to only return dataframes rather than specific variables. It should cut out a good amount of code.

**Pull request closed without merging**